javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,379 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named addAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  const newarray=[];
+  arr.forEach( (str) => { newarray.push(callback(str));});
+  return newarray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function called sortNames that takes an array of names and sorts them alphabetically. Capital letters should come before lowercase letters.
+
+For example: 'Cat' would come before 'apple'
+------------------------------------------------------------------------------------------------ */
+
+const sortNames = (arr) => {
+  let temp = ((arr.sort((a,b)=> a.toUpperCase() < b.toUpperCase() ? 1 : -1)));
+  return temp;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function called sortNumbers that takes an array of numbers and sorts them from smallest to largest.
+
+HINT: Beware... JS default is "Lexical" ordering.
+------------------------------------------------------------------------------------------------ */
+
+const sortNumbers = (arr) => {
+  let temp = arr.sort( (a, b) => {
+    return (a - b);});
+  return temp;
+};
+
```

Comment:
You don't NEED to name a variable to return. You could just

```js
const sortNumbers = (arr) => {
  return arr.sort((a, b) => a - b);
}
```
That said, what you have here is fine. The only thing I'd think about is renaming "temp" to something like "solution" or "answer" so that it's more obvious what your variable is for.